trade-finance-manager-ui/server/controllers/utilisation-reports/index.ts

```diff
@@ -1,30 +1,130 @@
+
+  const reportReconciliationSummary = getReportReconciliationSummaryViewModel(reconciliationSummaryApiResponse);
+  const reportPeriod = getFormattedReportPeriod();
+  const reportDueDate = await getFormattedReportDueDate(userToken);
+
+  return {
+    reportReconciliationSummary,
+    reportPeriod,
+    reportDueDate,
+    month: oneIndexedMonth,
+    year,
+  };
+};
 
 export const getUtilisationReports = async (req: Request, res: Response) => {
   const { userToken, user } = req.session;
 
   try {
-    const submissionMonth = getIsoMonth(new Date());
-    const reconciliationSummaryApiResponse = await api.getUtilisationReportsReconciliationSummary(submissionMonth, asString(userToken));
-
-    const reportReconciliationSummary = getReportReconciliationSummaryViewModel(reconciliationSummaryApiResponse);
-    const reportPeriod = getFormattedReportPeriod();
-    const reportDueDate = await getFormattedReportDueDate(userToken);
+    const { reportReconciliationSummary, reportPeriod, reportDueDate, month, year } = await getUtilisationReportsSummaryData(userToken);
 
     return res.render('utilisation-reports/utilisation-reports.njk', {
       user,
       activePrimaryNavigation: 'utilisation reports',
       reportReconciliationSummary,
       reportPeriod,
       reportDueDate,
+      month,
+      year,
     });
   } catch (error) {
     console.error('Error rendering utilisation reports page', error);
     return res.render('_partials/problem-with-service.njk', { user });
   }
 };
+
+const getReportIdentifiersFromUrl = (body: object, month: number, year: number): ReportIdentifier[] =>
+  Object.keys(body).reduce((reportIdentifiers, key) => {
+    const match = key.match(/(set-status--)(.*)(-)([a-f\d]+)/);
+    if (!match) {
+      return reportIdentifiers;
+    }
+
+    const idType = match.at(2);
+    const id = match.at(4);
+
+    if (!idType || !id) {
+      throw new Error('Failed to parse report identifiers from request body');
+    }
+
+    if (idType === 'reportId') {
+      return [...reportIdentifiers, { id }];
+    }
+
+    if (idType === 'bankId') {
+      return [...reportIdentifiers, { bankId: id, month, year }];
+    }
+
+    throw new Error('Failed to parse report identifiers from request body');
+  }, [] as ReportIdentifier[]);
+
+export type UpdateReportStatusPayload = {
+  reportsWithStatus: ReportWithStatus[];
+  user: TfmSessionUser;
+};
+
+export const updateUtilisationReportStatus = async (req: Request, res: Response) => {
+  const { user, userToken } = req.session;
+  const { status, month, year } = req.query;
+
+  try {
+    if (!user) {
+      throw new Error('User is not logged in');
+    }
+
+    const reportIdentifiers = getReportIdentifiersFromUrl(req.body as object, Number(month), Number(year));
+
+    if (status !== 'completed' && status !== 'not-completed') {
+      throw new Error('Supplied status instruction does not match any expected status');
```

Comment:
This error message is an example of how using `state` for `completed` and `not-completed` may avoid confusion